github_connector/models/github_repository_branch.py

```diff
@@ -192,7 +192,7 @@ def _download_code(self):
                     res = check_output(
                         ["git", "pull", "origin", branch.name], cwd=branch.local_path
                     )
-                    if branch.state == "to_download" or b"up-to-date" not in res:
+                    if b"up to date" not in res:
                         branch.write(
                             {
                                 "last_download_date": datetime.today(),
```

Comment:
Seeing the rest of the code, this can be:

```suggestion
                    vals = {"last_download_date": datetime.today()}
                    if b"up to date" not in res:
                        vals["state"] = "to_analyze"
                    branch.write(vals)
```
and removing the extra bits after this.